src/reconcile/rec_row.c

```diff
@@ -972,25 +986,43 @@ __wt_rec_row_leaf(
              * previous key (it's a fast path for simple, prefix-compressed keys), or by building
              * the key from scratch.
              */
-            if (__wt_row_leaf_key_info(page, copy, NULL, &cell, &tmpkey->data, &tmpkey->size))
+            __wt_row_leaf_key_info(page, copy, NULL, &cell, &key_data, &key_size, &key_prefix);
+            if (key_data == NULL) {
+                if (__wt_cell_type(cell) != WT_CELL_KEY)
+                    goto slow;
+                kpack = &_kpack;
+                __wt_cell_unpack_kv(session, page->dsk, cell, kpack);
+                key_data = kpack->data;
+                key_size = kpack->size;
+                key_prefix = kpack->prefix;
+            }
+            if (key_prefix == 0) {
+                tmpkey->data = key_data;
+                tmpkey->size = key_size;
                 goto build;
+            }
 
-            kpack = &_kpack;
-            __wt_cell_unpack_kv(session, page->dsk, cell, kpack);
-            if (kpack->type == WT_CELL_KEY && tmpkey->size >= kpack->prefix && tmpkey->size != 0) {
-                /*
-                 * Grow the buffer as necessary, ensuring data data has been copied into local
-                 * buffer space, then append the suffix to the prefix already in the buffer.
-                 *
-                 * Don't grow the buffer unnecessarily or copy data we don't need, truncate the
-                 * item's data length to the prefix bytes.
-                 */
-                tmpkey->size = kpack->prefix;
-                WT_ERR(__wt_buf_grow(session, tmpkey, tmpkey->size + kpack->size));
-                memcpy((uint8_t *)tmpkey->mem + tmpkey->size, kpack->data, kpack->size);
-                tmpkey->size += kpack->size;
-            } else
+            if (tmpkey->size == 0 || tmpkey->size < key_prefix)
+                goto slow;
+
+            /*
+             * Grow the buffer as necessary, ensuring data has been copied into local buffer space,
+             * then append the suffix to the prefix already in the buffer.
+             *
+             * Don't grow the buffer unnecessarily or copy data we don't need, truncate the item's
+             * CURRENT data length to the prefix bytes before growing the buffer.
```

Comment:
Same comment here as earlier about whether you intend to leave `CURRENT` in all caps.

Reply:
Yeah, `CURRENT` is deliberate.